Core/src/org/sleuthkit/autopsy/datamodel/FileTypesByExtension.java

```diff
@@ -340,24 +342,43 @@ long calculateChildCount() throws TskCoreException {
         }
     }
 
-    private static String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
-        StringBuilder query = new StringBuilder();
-        query.append("(dir_type = ").append(TskData.TSK_FS_NAME_TYPE_ENUM.REG.getValue()).append(")"); //NON-NLS
-        if (UserPreferences.hideKnownFilesInViewsTree()) {
-            query.append(" AND (known IS NULL OR known != ").append(TskData.FileKnown.KNOWN.getFileKnownValue()).append(")"); //NON-NLS
-        }
-        query.append(" AND (NULL"); //NON-NLS
-        for (String s : filter.getFilter()) {
-            query.append(" OR LOWER(name) LIKE LOWER('%").append(s).append("')"); //NON-NLS
+    private String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
+        if (filter.getFilter().isEmpty()) {
+            // We should never be given a search filter without extensions
+            // but if we are it is clearly a programming error so we throw 
+            // an IllegalArgumentException.
+            throw new IllegalArgumentException("Empty filter list passed to createQuery()"); // NON-NLS
+        }
+
+        String query = "(dir_type = " + TskData.TSK_FS_NAME_TYPE_ENUM.REG.getValue() + ")"
+                + (UserPreferences.hideKnownFilesInViewsTree() ? " AND (known IS NULL OR known != "
+                + TskData.FileKnown.KNOWN.getFileKnownValue() + ")" : " ")
+                + " AND (NULL "; //NON-NLS
+
+        if (skCase.getDatabaseType().equals(TskData.DbType.POSTGRESQL)) {
+            // For PostgreSQL we get a more efficient query by using builtin
+            // regular expression support and or'ing all extensions. We also
+            // escape the dot at the beginning of the extension.
+            // We will end up with a query that looks something like this:
+            // OR LOWER(name) ~ '(\.zip|\.rar|\.7zip|\.cab|\.jar|\.cpio|\.ar|\.gz|\.tgz|\.bz2)$')
+            query += "OR LOWER(name) ~ '(\\";
+            query += StringUtils.join(filter.getFilter().stream()
+                    .map(String::toLowerCase).collect(Collectors.toList()), "|\\");
```

Comment:
This is pretty nit picky, but I would use collectors.joining instead of stringutils.join, since there is already a stream involved.